src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -2477,33 +2480,20 @@ private void login(String primary, String primaryInstanceName, int primaryPortNu
-                                                                                            // user account locked
-                        || (SQLServerException.DRIVER_ERROR_INVALID_TDS == sqlex.getDriverErrorCode()) // invalid TDS
-                                                                                                       // received from
-                                                                                                       // server
-                        || (SQLServerException.DRIVER_ERROR_SSL_FAILED == sqlex.getDriverErrorCode()) // failure
-                                                                                                      // negotiating SSL
-                        || (SQLServerException.DRIVER_ERROR_INTERMITTENT_TLS_FAILED == sqlex.getDriverErrorCode()) // failure
-                                                                                                                   // TLS1.2
-                        || (SQLServerException.DRIVER_ERROR_UNSUPPORTED_CONFIG == sqlex.getDriverErrorCode()) // unsupported
-                                                                                                              // configuration
-                                                                                                              // (e.g.
-                                                                                                              // Sphinx,
-                                                                                                              // invalid
-                                                                                                              // packet
-                                                                                                              // size,
-                                                                                                              // etc.)
-                        || (SQLServerException.ERROR_SOCKET_TIMEOUT == sqlex.getDriverErrorCode()) // socket timeout
-                                                                                                   // ocurred
-                        || timerHasExpired(timerExpire)// no more time to try again
-                        || (state.equals(State.Connected) && !isDBMirroring)
-                // for non-dbmirroring cases, do not retry after tcp socket connection succeeds
-                ) {
-
+                int errorCode = sqlex.getErrorCode();
+                int driverErrorCode = sqlex.getDriverErrorCode();
+                if (SQLServerException.LOGON_FAILED == errorCode // actual logon failed, i.e. bad password
+                        || SQLServerException.PASSWORD_EXPIRED == errorCode // user account locked
+                        || SQLServerException.DRIVER_ERROR_INVALID_TDS == driverErrorCode // invalid TDS
```

Comment:
`SQLServerException.USER_ACCOUNT_LOCKED == sqlex.getErrorCode()` was removed, is this intentional?